### DISCUSSION TITLE
HTMX?

### DISCUSSION THREAD

**Discussion by martin-braun** [Instawork/hyperview#695] [Ideas]
I learned about hyperview through the hype of HTMX. I love the approach from HTMX and asked myself if there could be something similar for mobile apps. Taking hyperview into consideration I realized UI cannot be hybrid, I wonder if a transition layer is possible to blend HTMX with HXML. 

The idea is to be able to write one backend to serve the web and mobile. Are the concepts similar enough to approach such a thing? Where would a translation layer reside? I assume the best direction would be HXML->HTMX, but where and how could this be implemented?

**Comment by adamstep** [comment]
Hi @martin-braun . It is possible to have one backend to serve both the web and mobile versions, using HTMX templates for the web, and HXML templates for mobile. There's a section in our book [Hypermedia Systems](https://hypermedia.systems/) about this topic, check out https://hypermedia.systems/building-a-contacts-app-with-hyperview/#_one_backend_multiple_hypermedia_formats

If you truly want to write the app once (including templates) and run on both mobile and web, there's another option. Since Hyperview is built on top of React Native, you can use the [React Native for Web](https://necolas.github.io/react-native-web/docs/#:~:text=React%20Native%20for%20Web%20is,code%20in%20a%20web%20browser.) library to render your Hyperview app in HTML. We've experimented with this approach, there are a bunch of gotchas to have a nice-feeling web app, but it is possible.

**Reply by martin-braun** [reply]
@adamstep Thanks for your response, really appreciate it. Yes, the idea is to have one template to serve both platforms. I read somewhere about Hyperview using React Native under the hood, but for some reason this possibility went over my head. Unfortunately, even when succeeding with this, it would not allow to ship HTMX, as React Native for Web will send a React app to the client to render, but the Hyperview approach is very similar to HTMX, although I need to dig deeper. But, this is promising and I will dig deeper in the future to setup an interesting tech stack using Hyperview and Strapi.

Mind sharing your experiments so I can learn from them? :)

**Reply by martin-braun** [reply]
Strapi is not fulfilling, but Directus is. HTMX is an additional stack, I don't want to deal with right now. I kinda want to attempt this now: HXML with Expo targeting Android, iOS and Web and also having a proper nodejs backend alongside the project to have the client relevant stuff self-contained. I will report back my personal impressions. Thanks.
